src/config/ConfigurationInterface.h

```diff
@@ -1,7 +1,14 @@
 #ifndef CONFIGURATION_INTERFACE_H
 #define CONFIGURATION_INTERFACE_H
 
+namespace config {
+template <typename ConfigurationFileType>
 class ConfigurationInterface {
+public:
+    virtual ConfigurationFileType ReadConfigurationFile() = 0;
+    virtual int WriteConfigurationFile(ConfigurationFileType jsonfile) = 0;
+    virtual ~ConfigurationInterface() = default;
```

Comment:
One important method is `GetConfiguration` (or other name for it haha) which is responsible to return to us the value of the config and would be something like:
```cpp
int GetConfiguration(EnumType ConfigName)
```
That `int` is just an example, we'll do something really abstract with [std::variant](https://en.cppreference.com/w/cpp/utility/variant) to handle all type of different types. But don't worry for now, just add `int` or something simple.